extensions/amp-analytics/0.1/instrumentation.js

```diff
@@ -212,6 +204,97 @@ export class InstrumentationService {
   }
 
   /**
+   * Creates listeners for visibility conditions or calls the callback if all
+   * the conditions are met.
+   * @param {!AnalyticsEventListenerDef} The callback to call when the event
+   *   occurs.
+   * @param {!JSONObject} config Configuration for instrumentation.
+   * @private
+   */
+  createVisibilityListener_(callback, config) {
+    if (!this.isVisibilitySpecValid_(config)) {
+      return;
+    }
+
+    // TODO(avimehta, #1297): Add all the listeners so that visibility
+    // conditions are monitored and callback is called when the conditions
+    // are met.
+    if (this.viewer_.isVisible()) {
+      callback(new AnalyticsEvent(AnalyticsEventType.VISIBLE));
+    } else {
+      this.viewer_.onVisibilityChanged(() => {
+        if (this.viewer_.isVisible()) {
+          callback(new AnalyticsEvent(AnalyticsEventType.VISIBLE));
+        }
+      });
+    }
+  }
+
+  /**
+   * Checks and outputs information about visibilitySpecValidation.
+   * @param {!JSONObject} config Configuration for instrumentation.
+   */
+  isVisibilitySpecValid_(config) {
+    if (!config['visibilitySpec'] || !this.isViewabilityExperimentOn_()) {
+      return true;
+    }
+
+    const spec = config['visibilitySpec'];
+    if (!spec['selector'] || spec['selector'][0] != '#') {
+      user.error(this.TAG_, 'Visibility spec requires an id selector');
+      return false;
+    }
+
+    const ctMax = spec['continuousTimeMax'];
+    const ctMin = spec['continuousTimeMin'];
+    const ttMax = spec['totalTimeMax'];
+    const ttMin = spec['totalTimeMin'];
+
+    // "", 1, 0, undefined, 100, 101 are positive. -1, NaN are not.
+    function isPositiveNumber(num) {
+      return num === undefined || Math.sign(num) >= 0;
+    }
+
+    if (!isPositiveNumber(ctMin) || !isPositiveNumber(ctMax) ||
+        !isPositiveNumber(ttMin) || !isPositiveNumber(ttMax)) {
+      user.error(this.TAG_, 'Timing conditions should be positive integers ' +
+          'when specified.');
+      return false;
+    }
+
+    if ((ctMax || ttMax) && !spec['unload']) {
```

Comment:
if the min is 1.5 seconds, max is 2 seconds and there is not unload condition, the trigger will fire at 1.5 seconds. 

if min is 1.5 seocnds, max is 2 seconds and there is a unload condition, the trigger will only fire at unload and will make sure that the element was visible for less than 2 seconds. If it was visible for more than 2 seconds, the trigger will not fire. 

Note that this might not be useful to count a views but more important to count early exists (spam detection, bounces etc)...